cmd/fleet/serve.go

```diff
@@ -263,6 +263,35 @@ the way that the Fleet server works.
 				}
 			}
 
+			if config.Packaging.GlobalEnrollSecret != "" {
+				secrets, err := ds.GetEnrollSecrets(cmd.Context(), nil)
+				if err != nil {
+					initFatal(err, "loading enroll secrets")
+				}
+
+				var globalEnrollSecret string
+				for _, secret := range secrets {
+					if secret.TeamID == nil {
+						globalEnrollSecret = secret.Secret
+						break
+					}
+				}
+
+				if globalEnrollSecret != "" && globalEnrollSecret != config.Packaging.GlobalEnrollSecret {
```

Comment:
Sorry for the nit 🙈 
```
if globalEnrollSecret != "" {
  if globalEnrollSecret != config.Packaging.GlobalEnrollSecret {
    // print warning
  }
} else {
  // insert enroll secret
}
```

To avoid deletes+inserts in restarts.

Reply:
you're right, sorry about that. Fixed